### DISCUSSION TITLE
What does "Treat as source" do?

### DISCUSSION THREAD

**Discussion by nategreen** [six7/figma-tokens#726] [Q&A] [answered]
When I right-click a token set, there's a true/false option "Treat as source." Checking this changes the checkbox to an indeterminate state (I think?) but there's no other obvious change to the behavior or JSON. 🤔

![screenshot](https://user-images.githubusercontent.com/6674698/169304797-8a1b6a62-4e24-43bc-8d51-b07ebe3b028f.png)

**Comment by six7** [comment]
Treat as source makes this token set be used to resolve tokens but not use it to create any styles.

Think of a Dark theme library file where you don't want your global tokens be created. Until we had that feature users had to manually remove those groups after creating.

**Comment by iamnbutler** [comment, marked as answer]
> Right-click a set to enable treat as source. When a token set is treated as a source, these tokens will not be exported as styles but only used as a reference. You recognise a token set is marked as source by the dot inside the checbox.

Not sure if this was available at the time of the question, but this is now in the docs: https://docs.figmatokens.com/themes/token-sets#treat-as-source
